### PR TITLE
Fix #9875 SugarFeed shows 0 seconds ago and negative interval for cer…

### LINKED ISSUE #9875: SugarFeed shows 0 seconds ago and negative interval for certain datetime formats
#### Issue

SugarFeed shows 0 seconds ago and negative interval e.g -8 Hours -39 Minutes ago for certain users

#### Expected Behavior

SugarFeed in Dashlet should show proper time ago

#### Actual Behavior

SugarFeed shows incorrect time ago if date format is 23/12/2022

#### Steps to Reproduce

1.
2.
3.
4.

#### Your Environment

* SuiteCRM Version used: 7.13.0
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Chrome 108.0.5359.125 (Official Build) (64-bit)
* Environment name and version (e.g. MySQL, PHP 7): php 7.4
* Operating System and version (e.g Ubuntu 16.04): Windows 10

### PR DESCRIPTION
**Pull request by abuzarfaris**
…tain datetime formats

Use proper user datetime format to convert datetime string to timestamp in SugarFeed.php

## Description
use date_create_from_format with current user datetime format instead of strtotime because strtotime to time does lots of assumtions e.g  if the separator is a slash (/), then the American m/d/y is assumed. If the separator is a dash (-) or a dot (.), then the European d-m-y format is assumed

## Motivation and Context

SugarFeeds shows correctly for all users with different datetime format settings
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by pgorod** [comment]
Thanks! 
May I suggest some error handling on the return of **`date_create_from_format`**?

It's an alias of [DateTime::createFromFormat](https://www.php.net/manual/en/datetime.createfromformat.php)
```
Return Values

Returns a new DateTime instance or false on failure.
```
We should probably check if the function returned successfully...

**Comment by abuzarfaris** [comment]
> Thanks! May I suggest some error handling on the return of **`date_create_from_format`**?
> 
> It's an alias of [DateTime::createFromFormat](https://www.php.net/manual/en/datetime.createfromformat.php)
> 
> ```
> Return Values
> 
> Returns a new DateTime instance or false on failure.
> ```
> 
> We should probably check if the function returned successfully...
I have added proper logging incase startdate is not parsed

**Comment by pgorod** [comment]
Thanks @abuzarfaris !

**Comment by serhiisamko091184** [comment]
Hello @abuzarfaris 
thanks for your contribution, 

would you be so kind to squash commits in your branch into one (it means to combine multiple commits into single commit)?
Thanks a lot in advance.

Regards,
Serhii

**Comment by abuzarfaris** [comment]
> Hello @abuzarfaris thanks for your contribution,
> 
> would you be so kind to squash commits in your branch into one (it means to combine multiple commits into single commit)? Thanks a lot in advance.
> 
> Regards, Serhii

I have tried to squash can you check if it's correct

**Comment by abuzarfaris** [comment]
> Hi @abuzarfaris, thank you once again.
> 
> There shouldn't be any `Merge commits` like the following. Because we use rebasing instead of merging (which gives a way cleaner and easier to manage history timeline)
> 
> image

Fixed the merge issue and squashed the commits

File: modules/SugarFeed/SugarFeed.php [hunk 1]
```diff
@@ -476,9 +476,26 @@ public static function getTimeLapse($startDate)
...
+		}
+		
+        $second=date_create_from_format($user_format,$startDate);
+		if(empty($second)){
+			LoggerManager::getLogger()->warn('SugarFeed getTimeLapse: Could not fetch startDate ');
+			$second=0;
+		}
+		else{
+			$second=$second->getTimestamp();
+		}
```
Comment (clemente-raposo) [inline modules/SugarFeed/SugarFeed.php:498]: Hi @abuzarfaris, thank you for your PR and for the update.

Just  one minor thing, could you fix indentation on the `if / else` statements please?

Comment (clemente-raposo) [reply modules/SugarFeed/SugarFeed.php]: Hi @abuzarfaris, thank you

**Review by clemente-raposo — APPROVED** [review]

**Review by johnM2401 — APPROVED** [review]
LGTM, Checked each Time and Date Format, appears to as though Hours/Minutes renders as expected.

**Pull request merged**